src/credentials/examples/DeviceAttestationVerifierExample.cpp

```diff
@@ -0,0 +1,221 @@
+                0xCC, 0x52, 0x02, 0x21, 0x00, 0xA0, 0x36, 0xB0, 0xAF, 0x83, 0xE3, 0x45, 0x5C, 0x09, 0x38, 0xB9, 0x4B, 0xFF, 0x47,
+                0x75, 0x19, 0x72, 0xB8, 0xDC, 0xBE, 0x71, 0x9C, 0x9F, 0x61, 0xB5, 0x56, 0xEC, 0xB4, 0xEA, 0x17, 0xCC, 0x57 },
+              { 0xE7, 0xEF, 0xEA, 0xC3, 0x33, 0x5C, 0x5F, 0xD0, 0xC3, 0xE6,
+                0x34, 0x52, 0x9F, 0x16, 0x70, 0x46, 0xC4, 0xBC, 0xA5, 0x5C } },
+        };
+
+    size_t paaLookupTableIdx;
+    for (paaLookupTableIdx = 0; paaLookupTableIdx < ArraySize(sPAALookupTable); ++paaLookupTableIdx)
+    {
+        if (skid.data_equal(ByteSpan(sPAALookupTable[paaLookupTableIdx].mSKID)))
+        {
+            break;
+        }
+    }
+
+    VerifyOrReturnError(paaLookupTableIdx < ArraySize(sPAALookupTable), CHIP_ERROR_INVALID_ARGUMENT);
+
+    return CopySpanToMutableSpan(ByteSpan{ sPAALookupTable[paaLookupTableIdx].mPAACertificate }, out_dac_buffer);
+}
+
+class ExampleDACVerifier : public DeviceAttestationVerifier
+{
+public:
+    AttestationVerificationResult
+    VerifyAttestationInformation(const ByteSpan & attestation_info_buffer, const ByteSpan & attestation_challenge_buffer,
+                                 const ByteSpan & attestation_signature_buffer, const ByteSpan & pai_cert_der_buffer,
+                                 const ByteSpan & dac_cert_der_buffer, const ByteSpan & attestation_nonce) override;
+};
+
+AttestationVerificationResult ExampleDACVerifier::VerifyAttestationInformation(const ByteSpan & attestation_info_buffer,
+                                                                               const ByteSpan & attestation_challenge_buffer,
+                                                                               const ByteSpan & attestation_signature_buffer,
+                                                                               const ByteSpan & pai_cert_der_buffer,
+                                                                               const ByteSpan & dac_cert_der_buffer,
+                                                                               const ByteSpan & attestation_nonce)
+{
+    // match DAC and PAI VIDs
+    if (!pai_cert_der_buffer.empty())
+    {
+        VendorId paiVid;
+        VendorId dacVid;
+
+        CHIP_ERROR error = ExtractVIDFromX509Cert(pai_cert_der_buffer, paiVid);
+        VerifyOrReturnError(error == CHIP_NO_ERROR || error == CHIP_ERROR_KEY_NOT_FOUND,
+                            AttestationVerificationResult::kPaiFormatInvalid);
+
+        if (error == CHIP_NO_ERROR)
+        {
+            VerifyOrReturnError(ExtractVIDFromX509Cert(dac_cert_der_buffer, dacVid) == CHIP_NO_ERROR,
+                                AttestationVerificationResult::kDacFormatInvalid);
+
+            VerifyOrReturnError(paiVid == dacVid, AttestationVerificationResult::kDacVendorIdMismatch);
+        }
+    }
+
+    P256PublicKey remoteManufacturerPubkey;
```

Comment:
There is a mix of styles in arguments and variables between `snake_case` and `camelCase`. I thought `snake_case` was common for variables, but it looks like we have a mix in the code base. Would be great not to mix in this file.

Reply:
done